hoomd/hpmc/tune/boxmc_move_size.py

```diff
@@ -0,0 +1,302 @@
+        self._should_update_tunables = False
+        self._tuned = 0
+
+    def _flag_move_size_update(self, value):
+        self._update_move_sizes = True
+        return value
+
+    def _flag_new_tunables(self, value):
+        self._should_update_tunables = True
+        return value
+        self._param_dict._cpp_obj = 5
+
+
+class BoxMCMoveSize(_InternalCustomTuner):
+    """Tunes ``BoxMC`` move sizes to targeted acceptance rate.
+
+    For most common creation of a `BoxMCMoveSize` tuner see
+    `BoxMCMoveSize.secant_solver` and `BoxMCMoveSize.scale_solver`
+    respectively.
+
+    Args:
+        trigger (hoomd.trigger.Trigger): ``Trigger`` to determine when to run
+            the tuner.
+        boxmc (hoomd.hpmc.update.BoxMC): The `hoomd.hpmc.update.BoxMC` object to
+            tune.
+        moves (list[str]): A list of types of moves to tune. Available options
+            are 'delta'.
+        target (float): The acceptance rate for trial moves that is desired. The
+            value should be between 0 and 1.
+        solver (`hoomd.tune.SolverStep`): A solver that tunes move sizes to
+            reach the specified target.
+        types (list[str]): A list of string particle types to tune the move
+            size for, defaults to None which upon attaching will tune all types
+            in the system currently.
+        max_move_size (float): The maximum volume move size to attempt.
+
+    Attributes:
+        trigger (hoomd.trigger.Trigger): ``Trigger`` to determine when to run
+            the tuner.
+        boxmc (hoomd.hpmc.update.BoxMC): The `hoomd.hpmc.update.BoxMC` object to
+            tune.
+        moves (list[str]): A list of types of moves to tune. Available options
+            are 'a' and 'd'.
+        target (float): The acceptance rate for trial moves that is desired. The
+            value should be between 0 and 1.
+        solver (hoomd.tune.SolverStep): A solver that tunes move sizes to
+            reach the specified target.
+        types (list[str]): A list of string particle
+            types to tune the move size for, defaults to None which upon
+            attaching will tune all types in the system currently.
+        max_move_size (float): The maximum value of a volume move
+            size to attempt.
+
+    Warning:
+        Over-limiting the maximum move sizes can lead to the inability to
+        converge to the desired acceptance rate.
+    """
+    _internal_class = _InternalBoxMCMoveSize
+
+    @classmethod
+    def scale_solver(cls,
+                     trigger,
+                     boxmc,
+                     moves,
+                     target,
+                     max_move_size=None,
+                     max_scale=2.,
+                     gamma=1.,
+                     tol=1e-2):
+        """Create a `BoxMCMoveSize` tuner with a `hoomd.tune.ScaleSolver`.
+
+        Args:
+            trigger (hoomd.trigger.Trigger): ``Trigger`` to determine when to
```

Comment:
Correct this documentation, it appears to be copied and pasted from the mc integrator move size tuner.